### ISSUE TITLE
Suggestion: Add `Model.exists(...)`

### ISSUE THREAD

**Issue by SimonSchick** [sequelize/sequelize#10187]
## What are you doing?

Rationale: Most people (myself included at times) use `findOne()` to check if a certain record exists, this also pulls data from the db and constructors models, calls hooks etc. and generally involves LOTS of overhead.

I currently have a sample implementation that obviously doesn't address all issues mentioned above 
but at least doesn't pull the whole record and doesn't create a model.
(behold typescript):
```ts
  public static async exists(options: NonNullFindOptions): Promise<true>;
  public static async exists(options: FindOptions): Promise<boolean>;
  public static async exists(options: FindOptions | NonNullFindOptions): Promise<boolean> {
    options.attributes = [this.primaryKeyAttribute];
    options.raw = true;
    return !!(await this.findOne(options));
  }
```

## What do you expect to happen?
I'd really like a method like this in sequelize by default since checking if a record exists without needing the data seems like a very common use case.

## What is actually happening?
The method doesn't exist :c

**Comment by frlinw** [comment]
What about [count](https://github.com/sequelize/sequelize/blob/master/lib/model.js#L1978) ?

**Comment by SimonSchick** [comment]
Ideally it would execute something like `select 1 from where ...`, not run an aggregation.

**Comment by SimonSchick** [comment]
@knoxcard you missed the entire point.
This is to check if a single column/model exists in the database without pulling it and constructing a model instance etc., generally avoiding a bunch of runtime overhead.

This is NOT to check if a model class exists (or whatever you thought it was).

**Comment by SimonSchick** [comment]
The entire point of this is to perform something like `select 1 where ...` and entirely skip creating the model instance etc. just returning a boolean, that way you don't need to specify any columns or anything.

**Comment by knoxcard** [comment]
Okay think I finally understand, sounds like the only solution for now is creating a function that executes that raw SQL query in Sequelize.

**Comment by ustun** [comment]
Not stale

**Comment by Betree** [comment]
Postgres is actually able to make really nice optimizations when using `EXISTS`.

An EXPLAIN ANALYZE on this query:

```sql
SELECT 1
FROM "Transactions" AS "Transaction"
WHERE ("Transaction"."deletedAt" IS NULL AND
     ("Transaction"."CollectiveId" = 479118 OR "Transaction"."FromCollectiveId" = 479118))
```

Results in (cost=19395.70..19395.70 rows=1 width=8) (actual time=88.933..88.935 rows=1 loops=1)

The exact same equivalent with `EXISTS`:

```sql
SELECT EXISTS (
  SELECT 1
  FROM "Transactions" AS "Transaction"
  WHERE ("Transaction"."deletedAt" IS NULL AND
       ("Transaction"."CollectiveId" = 479118 OR "Transaction"."FromCollectiveId" = 479118))
)
```

Results in (cost=702.31..702.32 rows=1 width=1) (actual time=0.025..0.026 rows=1 loops=1) - the cost has been divided by almost 30!

I would love to be able to achieve the same thing with sequelize:

```es6
return models.Transaction.exists({ where: { ... } })
```

@papb Is this issue still in your pipe?

**Comment by ephys** [comment]
Pedro stopped contributing to Sequelize but it's still a feature we'd like to get to eventually. Unfortunately it's not a high priority for us right now, as we're focusing on fixing The Bad Parts

**Comment by Betree** [comment]
@ephys understandable! I will be happy to contribute to the code and/or specs when there will be more bandwidth to prioritize it.

**Comment by ephys** [comment]
We're open to reviewing a PR if you want to give it a try :)

**Comment by fzn0x** [comment]
> We're open to reviewing a PR if you want to give it a try :)

Maybe we can add `hacktoberfest` as well, if we want to give the bonus for everyone working for this issue.

**Comment by WikiRik** [comment]
We have added `hacktoberfest` to the repo so all approved PRs count towards the goal. But adding the label to the issue would increase the visibility of it

**Comment by fzn0x** [comment]
> We have added `hacktoberfest` to the repo so all approved PRs count towards the goal. But adding the label to the issue would increase the visibility of it

Thanks for the information, I thought we must added it.

**Comment by Pulkit0729** [comment]
Is this still an open issue?

**Comment by WikiRik** [comment]
Yes, this is still open as far as I know. Just like mentioned in October, a PR from the community is welcome. Initial implementation can do the following; https://github.com/sequelize/sequelize/issues/10187#issuecomment-1277384989

**Comment by erickbRodrigues** [comment]
That would be really nice to have!

**Comment by elizatlawy** [comment]
I would also love to see this feature.

**Comment by klondikemarlen** [comment]
I'd like to see something like
```ts
export abstract class Model<....> {
  public static async exists<M extends Model>(
    this: ModelStatic<M>,
    options?: Omit<CountOptions<Attributes<M>>, "group">
  ): Promise<boolean> {
    const count = await this.count(options)
    return count > 0
  }
}
```

but even more than that, I'd like to see some examples of how to extend the base model instance, so I can add these kind of things myself, until they become part of the standard library.

e.g.
```ts
export abstract class BaseModel<
  // eslint-disable-next-line @typescript-eslint/ban-types, @typescript-eslint/no-explicit-any
  TModelAttributes extends {} = any,
  // eslint-disable-next-line @typescript-eslint/ban-types
  TCreationAttributes extends {} = TModelAttributes,
> extends Model<TModelAttributes, TCreationAttributes> {
  public static async exists<M extends BaseModel>(
    this: ModelStatic<M>,
    options?: Omit<CountOptions<Attributes<M>>, "group">
  ): Promise<boolean> {
    const count = await this.count(options)
    return count > 0
  }
}
```

Sort of works, until you try to chain methods methods like
```ts
const isAccessible = await Dataset.scope([
  {
    method: ["accessibleViaAccessGrantsBy", this.user],
  },
  {
    method: ["withApprovedAccessRequestsFor", this.user],
  },
]).exists()
```

If anyone is curious why I want something that seems so trivial, its because this code look horrible to me.
```ts
const isAccessible =
      (await Dataset.scope([
        {
          method: ["accessibleViaAccessGrantsBy", this.user],
        },
        {
          method: ["withApprovedAccessRequestsFor", this.user],
        },
      ]).count()) > 0
```
